File: backend/models/directory.go
package models

import (
	"time"

	"github.com/globalsign/mgo/bson"
)

//Directory - directroy structure
type Directory struct {
	Title     string        `vd:"len($)>0; msg:'Title required'" json:"title" bson:"title"`
	Removed   bool          `json:"removed" bson:"removed"`
	CreatedBy string        `json:"-" bson:"createdBy"`
	ID        bson.ObjectId `json:"id" bson:"_id"`
	When      time.Time     `json:"when,omitempty" bson:"when,omitempty"`
	ParentID  string        `vd:"len($)>0; msg:'ParentId required'" json:"parentId" bson:"parentId"`
}

File: src/app/core/models/directory.ts
export class Directory {
    title: string
    removed: boolean
    id: string
    when: Date
    parentId: string

    static fromJson(json: any = {}): Directory {
        const dir = new Directory()
        dir.title = json.title
        dir.removed = json.removed
        dir.id = json.id
        dir.parentId = json.parentId
        dir.when = json.when
        return dir
    }
}


